docs/throughs/take.md

```diff
@@ -1,13 +1,55 @@
 # pull-stream/throughs/take
 
-## usage
+## Example usage
 
-### `take = require('pull-stream/throughs/take')`
+```js
+var pull = require('pull-stream')
+// var take = require('pull-stream/throughs/take') // if you just need take
 
-### `take(test [, opts])`
+pull(
+  pull.values(['a', 'b', 'c', 'd', 'e']),
+  pull.take(3),
+  pull.collect((err, data) => {
+    console.log(data)
+    // => ['a', 'b', 'c']
+  })
+)
+```
 
-If test is a function, read data from the source stream and forward it downstream until test(data) returns false.
+## API
 
-If `opts.last` is set to true, the data for which the test failed will be included in what is forwarded.
+take has 2 valid signatures: 
 
-If test is an integer, take n item from the source.
+### `take(n) => through`
+
+Where `n` is a positive integer.
+`take` pulls n values from the source and then closes the stream.
+This is really useful for limiting how much you pull.
+
+### `take(testFn [, opts]) => through`
+
+If `testFn` is a function, read data from the source stream and forward it downstream until `testFn(data)` returns false, then close the stream.
+
+`opts` is an optional Object of form `{ last: Boolean }`, where `opts.last` determines whether the last value tested (before closing the stream) is included or excluded (default). e.g.
+
+```js
+pull(
+  pull.values([1, 2, 3, 4, 5, 6, 7, 8, 9, 10]),
+  pull.take(n => n < 4.6), { last: true }),  // include the last value tested (5)
+  pull.collect(function (err, results) {
+    console.log(results)
+    // => [1, 2, 3, 4, 5]
+  })
+})
+```
+
+```js
+pull(
+  pull.values([1, 2, 3, 4, 5, 6, 7, 8, 9, 10]),
+  pull.take(n => n < 4.6), { last: false }),  // exclude the last value test (5)
```

Comment:
maybe here don't use `last: false` (but keep the comment) to show that by default it's false.

Reply:
could do, I'm going to leave it as is, because using this `testFn` isn't used much as far as I know.
I think they should be seperated into : 
- `pull.take(n)`
- `pull.till(testFn, opts)`

but that's another PR